### DISCUSSION TITLE
dynamicImport loading问题

### DISCUSSION THREAD

**Discussion by lzw10168** [umijs/umi#9948] [Q&A] [answered]
* 问题描述: 目前项目是启用了dynamicImport, 也配置了loading组件. 在跳转路由加载资源的时候也能正确显示loading. 但是如果资源在有缓存的情况下, loading效果就会一闪而过. 
* 期望处理方法: 判断一个时间点, 比如50ms. 100ms, 资源回来了,再跳转路由,  就不显示loading了, 这样也不会有明显卡顿, 体验感上也会更好. 但是不知道umi有没有办法处理这个问题.  求助各位.

**Comment by fz6m** [comment, marked as answer]
在 loading 组件里条件渲染，自己判断下时间差。

```ts
  const [show, setShow] = useState(false)
  
  useEffect(() => {
    const id = setTimeout(() => {
      setShow(true)
    }, 500)
    return () => {  clear(id)  }
  }, [])

  if (show) {
    return <Loading />
  }

  return null
```
